base/abstractarraymath.jl

```diff
@@ -4,6 +4,15 @@
 
 isreal(x::AbstractArray) = all(isreal,x)
 iszero(x::AbstractArray) = all(iszero,x)
+isone(x::AbstractVector) = false # vectors are not multiplicative identities
+function isone(x::AbstractMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    @inbounds for i in 1:m
```

Comment:
I think this should either just be `x == one(x)` or a loop through all elements to avoid allocating `one(x)`. It might be better to define the loop based version for `StridedMatrix` though since it will be very inefficient for `SparseMatrixCSC` and `DArray`.